packages/@aws-cdk/aws-servicecatalog/lib/private/association-manager.ts

```diff
@@ -74,6 +77,30 @@ export class AssociationManager {
     }
   }
 
+  public static constrainCloudFormationParameters(
+    portfolio: IPortfolio, product: IProduct,
+    options: ProvisioningRuleOptions,
+  ): void {
+    this.validateCommonConstraintOptions(portfolio, product, options);
```

Comment:
Let's refactor this to get rid this duplication.

Change `associateProductWithPortfolio()` to take a third, optional property, of type `CommonConstraintOptions`. If it's passed, it will call `validateCommonConstraintOptions()`.

Then, change the calls in these methods from `associateProductWithPortfolio(portfolio, product)` to `associateProductWithPortfolio(portfolio, product, options)`, and remove the `this.validateCommonConstraintOptions()` line.

Same comment in the other public methods of this class.

Reply:
sure, in that case also removing the `validate...` function and calling the method body directly.